### PR TITLE
(feat): custom signin page

### LINKED ISSUE #5856: [FEATURE] Improve login page design
### Description

Current login page design is not so descriptive, it's just a button there. Maybe add some texts and a logo to present it better. I've created a demo design in my local system. If it passes let me know, I would like to work on this 🙂

Screenshots below 👇

### Current version: 
image

### Proposed version:
image

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by NiazMorshed2007**
This feature closes #5856 by adding custom login page.

#### `1. Configuration part`
I added some configuration in the `[...nextauth].js` file to allow creating custom page in `auth/signin`. Changes I added 👇
```js
const authOptions = {
// ...
  pages: {
    signIn: "/auth/signin",
  }
};
```

#### `2. Design and connecting the login functionality part`
Kept the design simple but I'm always open to redesign if you wish!

###### Can I see a screenshot?
Sure! Here is how it's currently look like:
<br />
image

###### How did I managed the login funtionality?
I followed the best practices from NextAuth documentation to implement this. Here is a little brief about it 👇:

1. First of all I imported these necessary functions and modules
``` js
import { getServerSession } from "next-auth/next"
import { getProviders, signIn } from "next-auth/react"
import { authOptions } from "pages/api/auth/[...nextauth]"
```

2. And then with `getServerSideProps()` function I checked if the user is already logged in or not. If the user is logged I sent them to `/` page and if not I pushed the providers in the `props` of the page
```js
export async function getServerSideProps(context) {
    const session = await getServerSession(context.req, context.res, authOptions);

    // if the user is logged in redirect, (note: don't redirect to the same page otherwise it'll be in a infinite loop)
    if (session) {
        return { redirect: { destination: "/" } };
    }

    const providers = await getProviders();

    return {
        props: { providers: providers ?? [] },
    }
}
```

3. And then I grabbed the providers from the `props` and turned `providers` into an array using `Object.values()` and mapped over to display different auth providers. Notice the `handleProviderIcon` function here, it will check the provider name and render icon accordingly. 

```js
 {Object.values(providers).map((provider) => (
                    <button
                        key={provider.name}
                        onClick={() => signIn(provider.id)}
                        className="group relative transition-all flex gap-4 items-center w-full justify-center rounded-md bg-primary-medium p-3 text-sm font-medium text-white hover:bg-primary-medium/60 outline-none"
                    >
                        {handleProviderIcon(provider.name)} Continue with {provider.name}
                    </button>
 ))}
```

The `handleProviderIcon` function:

```js
    const handleProviderIcon = (provider_name) => {
        if (provider_name === "GitHub") {
            return <BsGithub className="text-xl" />
        }
    }
```

That's is the wrap.
I look forward for reviews. 
Thanks!🙌

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/6016"></a>

### CODE REVIEW

**Comment by ChinmayMhatre** [comment]
A couple of suggestion from my end : 
- Create the page for both dark mode and light mode by using the `:dark` variant ( let me know if you need any help with this )
- Take into consideration the contrast ratio between the background and foreground elements.
- Use the colours from the tailwind css config to keep consistency

**Review by iamthecloverly — APPROVED** [review]

**Comment by NiazMorshed2007** [comment]
> A couple of suggestion from my end 

Thanks @ChinmayMhatre for the suggestion, I'll work on this when I get free, also I may reach out to you on EddieHub discrod for help :)

**Comment by NiazMorshed2007** [comment]
@ChinmayMhatre I've done some tweaking. Can you review it?

**Comment by ChinmayMhatre** [comment]
> @ChinmayMhatre I've done some tweaking. Can you review it?

I'll take a look tommorow!

**Comment by ChinmayMhatre** [comment]
@NiazMorshed2007 please add screenshots of the page in light mode and dark mode

**Comment by NiazMorshed2007** [comment]
@ChinmayMhatre  Sure thing!

`The light version`:
image

`The dark version`:
image

**Comment by ChinmayMhatre** [comment]
Ui looks good to me. Maybe @eddiejaoude can take a look for any more optimizations.

File: pages/auth/signin.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    )
+}
+
+
+export async function getServerSideProps(context) {
+    const session = await getServerSession(context.req, context.res, authOptions);
+
+    // if the user is logged in redirect, (note: don't redirect to the same page otherwise it'll be in a infinite loop)
+    if (session) {
+        return { redirect: { destination: "/" } };
```
Comment (eddiejaoude) [inline pages/auth/signin.js:55]: I think if the user is logged in, we should redirect to the dashboard page, that we do after successful login https://github.com/EddieHubCommunity/LinkFree/pull/6016/files#diff-e0861d63eed5eacaa4b7c51e620e1da44a7579bccf244d96b26efb5d93c6b8a3L29

File: pages/auth/signin.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+SignIn.getLayout = function getLayout(page) {
+    return (
+        <BlankLayout>
+            {page}
+        </BlankLayout>
+    )
+}
+
+
+export async function getServerSideProps(context) {
```
Comment (eddiejaoude) [inline pages/auth/signin.js:50]: Please move `getServerSideProps` before the react component to be consistent with out other files

File: pages/auth/signin.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            </div>
+        </div>
+    )
+}
+
+SignIn.getLayout = function getLayout(page) {
+    return (
+        <BlankLayout>
+            {page}
+        </BlankLayout>
```
Comment (eddiejaoude) [inline pages/auth/signin.js:45]: The formatting doesn't look quite right, can you check you are using our Prettier config in VScode

File: pages/auth/signin.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import BlankLayout from "@components/layouts/BlankLayout"
+import { getServerSession } from "next-auth/next"
+import { getProviders, signIn } from "next-auth/react"
+import { authOptions } from "pages/api/auth/[...nextauth]"
+import { BsGithub } from "react-icons/bs"
+
+export default function SignIn({ providers }) {
+
+    const handleProviderIcon = (provider_name) => {
+        if (provider_name === "GitHub") {
```
Comment (eddiejaoude) [inline pages/auth/signin.js:10]: A `switch` would be better here and then add a `default` at the end for a fallback login icon

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Looks great 👍  some inline comments

File: pages/auth/signin.js (hunk 3 above)
Comment (NiazMorshed2007) [reply pages/auth/signin.js:45]: I've done the other requested changes. But before I commit I've the issue with prettier still. Is the configuration for prettier is the `.prettierrc` file? I got it empty

image

File: pages/auth/signin.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            return <BsGithub className="text-xl" />
+        }
+    }
+
+    return (
+        <div className="flex min-h-screen items-center justify-center px-4 py-12 sm:px-6 lg:px-8">
+            <div className="w-full max-w-sm space-y-8">
+                <img
+                    className="mx-auto h-16 w-auto"
+                    src="/logo512.png"
```
Comment (FrozenHearth) [inline pages/auth/signin.js:20]: Can we use Next's Image component here?

Comment (eddiejaoude) [reply pages/auth/signin.js:20]: Good spot 👍

File: pages/auth/signin.js (hunk 4 above)
Comment (eddiejaoude) [reply pages/auth/signin.js:10]: But for now maybe just add a return out side of the `if` with a default key/secure icon

File: pages/auth/signin.js (hunk 3 above)
Comment (eddiejaoude) [reply pages/auth/signin.js:45]: yep we use the default, that is fine, the plugin will work on save usually

File: pages/auth/signin.js (hunk 5 above)
Comment (NiazMorshed2007) [reply pages/auth/signin.js:20]: I'll fix it! Good find @FrozenHearth

File: pages/auth/signin.js (hunk 4 above)
Comment (NiazMorshed2007) [reply pages/auth/signin.js:10]: You mean like this?  

```js
const handleProviderIcon = (provider_name) => {
        if (provider_name === "GitHub") {
            return <BsGithub className="text-xl" />
        }

        return <LockIcon />
    }
```

File: pages/auth/signin.js (hunk 3 above)
Comment (NiazMorshed2007) [reply pages/auth/signin.js:45]: Yeah, but it doesn't fix the formatting. I've pinged on discord about this issue.

File: pages/auth/signin.js (hunk 1 above)
Comment (NiazMorshed2007) [reply pages/auth/signin.js:55]: Done.

File: pages/auth/signin.js (hunk 4 above)
Comment (eddiejaoude) [reply pages/auth/signin.js:29]: yes 👍 , but you might want to add the same class to it also

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**